lib/android-tools-info.ts

```diff
@@ -48,100 +43,54 @@ export class AndroidToolsInfo implements IAndroidToolsInfo {
-			detectedErrors = true;
-		}
 
-		if (!toolsInfoData.buildToolsVersion) {
-			const buildToolsRange = this.getBuildToolsRange();
-			const versionRangeMatches = buildToolsRange.match(/^.*?([\d\.]+)\s+.*?([\d\.]+)$/);
-			let message = `You can install any version in the following range: '${buildToolsRange}'.`;
-
-			// Improve message in case buildToolsRange is something like: ">=22.0.0 <=22.0.0" - same numbers on both sides
-			if (versionRangeMatches && versionRangeMatches[1] && versionRangeMatches[2] && versionRangeMatches[1] === versionRangeMatches[2]) {
-				message = `You have to install version ${versionRangeMatches[1]}.`;
-			}
+		detectedErrors = androidToolsInfo.validateInfo().map(warning => this.printMessage(warning.warning)).length > 0;
 
-			let invalidBuildToolsAdditionalMsg = `Run \`\$ ${this.getPathToSdkManagementTool()}\` from your command-line to install required \`Android Build Tools\`.`;
-			if (!isAndroidHomeValid) {
-				invalidBuildToolsAdditionalMsg += ' In case you already have them installed, make sure `ANDROID_HOME` environment variable is set correctly.';
-			}
-
-			this.printMessage("You need to have the Android SDK Build-tools installed on your system. " + message, invalidBuildToolsAdditionalMsg);
-			detectedErrors = true;
+		if (options && options.validateTargetSdk) {
+			detectedErrors = this.validateTargetSdk();
 		}
 
-		if (!toolsInfoData.supportRepositoryVersion) {
-			let invalidSupportLibAdditionalMsg = `Run \`\$ ${this.getPathToSdkManagementTool()}\` to manage the Android Support Repository.`;
-			if (!isAndroidHomeValid) {
-				invalidSupportLibAdditionalMsg += ' In case you already have it installed, make sure `ANDROID_HOME` environment variable is set correctly.';
-			}
-			this.printMessage(`You need to have Android SDK ${AndroidToolsInfo.MIN_REQUIRED_COMPILE_TARGET} or later and the latest Android Support Repository installed on your system.`, invalidSupportLibAdditionalMsg);
-			detectedErrors = true;
-		}
+		return detectedErrors || !isAndroidHomeValid;
+	}
 
-		if (options && options.validateTargetSdk) {
-			const targetSdk = toolsInfoData.targetSdkVersion;
-			const newTarget = `${AndroidToolsInfo.ANDROID_TARGET_PREFIX}-${targetSdk}`;
-			if (!_.includes(AndroidToolsInfo.SUPPORTED_TARGETS, newTarget)) {
-				const supportedVersions = AndroidToolsInfo.SUPPORTED_TARGETS.sort();
-				const minSupportedVersion = this.parseAndroidSdkString(_.first(supportedVersions));
-
-				if (targetSdk && (targetSdk < minSupportedVersion)) {
-					this.printMessage(`The selected Android target SDK ${newTarget} is not supported. You must target ${minSupportedVersion} or later.`);
-					detectedErrors = true;
-				} else if (!targetSdk || targetSdk > this.getMaxSupportedVersion()) {
-					this.$logger.warn(`Support for the selected Android target SDK ${newTarget} is not verified. Your Android app might not work as expected.`);
-				}
+	public validateTargetSdk(options?: { showWarningsAsErrors: boolean }): boolean {
```

Comment:
You can extract `{ showWarningsAsErrors: boolean }` as an interface.
While you're on it you can extract `{ validateTargetSdk: boolean}` too and reuse both everywhere